src/Inkluzitron/Modules/Vote/VoteService.cs

```diff
@@ -0,0 +1,331 @@
+                    messageReference: voteCommandReference
+                );
+
+                using var dbContext = DbFactory.Create();
+                dbContext.VoteReplyRecords.Add(new VoteReplyRecord
+                {
+                    GuildId = guildId,
+                    ChannelId = voteCommandReference.ChannelId,
+                    MessageId = voteCommandMessage.Id,
+                    ReplyId = newVoteReply.Id
+                });
+
+                await dbContext.SaveChangesAsync();
+            }
+            finally
+            {
+                ReplySemaphore.Release();
+            }
+        }
+
+        public async Task ProcessVoteCommandAsync(IUserMessage voteCommandMessage, string voteDefinitionText)
+        {
+            if (voteCommandMessage.Channel is not IGuildChannel guildChannel)
+                return;
+
+            var parse = await Parser.TryParse(new CommandContext(Client, voteCommandMessage), voteDefinitionText);
+
+            if (!parse.Success)
+            {
+                await UpdateVoteReplyAsync(voteCommandMessage, parse.ProblemDescription);
+                return;
+            }
+
+            var allEmotes = new HashSet<IEmote>(parse.Definition.Options.Keys);
+            allEmotes.UnionWith(voteCommandMessage.Reactions.Keys);
+            var failedEmotes = new List<IEmote>();
+            var summary = ComposeSummary(voteCommandMessage, parse.Definition);
+
+            foreach (var emote in allEmotes)
+            {
+                var shouldBePresent = parse.Definition.Options.ContainsKey(emote);
+                var isPresent = voteCommandMessage.Reactions.TryGetValue(emote, out var reaction);
+
+                if (!shouldBePresent && isPresent)
+                {
+                    await voteCommandMessage.RemoveAllReactionsForEmoteAsync(emote);
+                }
+                else if (shouldBePresent && !isPresent)
+                {
+                    try
+                    {
+                        await voteCommandMessage.AddReactionAsync(emote);
+                    }
+                    catch
+                    {
+                        failedEmotes.Add(emote);
+                    }
+                }
+            }
+
+            var tail = failedEmotes.Count == 0
+                ? string.Empty
+                : Environment.NewLine + string.Format(
+                    VoteTranslations.UnaccessibleEmotes,
+                    new FormatByValue(failedEmotes.Count),
+                    string.Join(", ", failedEmotes.Select(e => $"`{e}`"))
+                );
+
+            await UpdateVoteReplyAsync(voteCommandMessage, summary + tail);
+
+            if (parse.Definition.Deadline is DateTimeOffset votingDeadline)
+            {
+                var scheduledTaskTag = voteCommandMessage.GetJumpUrl();
+
+                foreach (var existingScheduledTask in await ScheduledTasksService.LookupAsync("EndOfVoting", scheduledTaskTag))
```

Comment:
To EndOfVoting vidím už podruhé. Co takhle na to udělat nějakou konstantu? :hmmnote:

Reply:
Udělal jsem z toho konstantu.